var/spack/repos/builtin/packages/py-keras/package.py

```diff
@@ -85,30 +87,28 @@ class PyKeras(PythonPackage):
         depends_on("py-absl-py", when="@2.6:")
         depends_on("py-numpy")
         depends_on("py-rich", when="@3:")
-        depends_on("py-namex@0.0.8:", when="@3.3.3:")
         depends_on("py-namex", when="@3:")
         depends_on("py-h5py")
         depends_on("py-optree", when="@3.1:")
         depends_on("py-ml-dtypes", when="@3.0.5:")
         depends_on("py-packaging", when="@3.4:")
 
         # requirements-common.txt
```

Comment:
The only text files showing up for `3.8.0` are:

```
./keras.egg-info/SOURCES.txt
./keras.egg-info/dependency_links.txt
./keras.egg-info/requires.txt
./keras.egg-info/top_level.txt
```

Reply:
Yep, these files are only packaged on GitHub, not in the PyPI tarball. Unfortunately keras doesn't formally document any of its optional dependencies, you have to scrape through these files and decide exactly which you want and what they do. Someday I may open an issue with them to make this better. For now, as long as no one complains, I think the current required deps are sufficient.